File: src/main/java/com/chg/yuaicodemother/kafka/TaskResultEvent.java
package com.chg.yuaicodemother.kafka;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Python AI 服务回传的任务结果事件，对应 task-result-topic 中的消息格式。
 * 字段名使用 camelCase（与 Python 端 Pydantic alias 一致）。
 *
 * @see com.chg.yuaicodemother.constant.kafkaConstant#TASK_RESULT_TOPIC
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TaskResultEvent {

    /**
     * 任务唯一标识，同时作为 Kafka 消息 key（保证同任务分到同一 partition）
     */
    @JsonProperty("taskId")
    private String taskId;

    /**
     * 应用 ID
     */
    @JsonProperty("appId")
    private Long appId;

    /**
     * 用户 ID
     */
    @JsonProperty("userId")
    private Long userId;

    /**
     * 任务状态：STREAMING / SUCCESS / FAILED / INTERRUPTED
     */
    @JsonProperty("status")
    private String status;

    /**
     * 消息类型：ai / user / system
     */
    @JsonProperty("messageType")
    private String messageType;

    /**
     * 本次生成的文本片段
     */
    @JsonProperty("content")
    private String content;

    /**
     * 分片序号，从 0 开始递增，用于幂等 key 和前端排序
     */
    @JsonProperty("seq")
    private Integer seq;

    /**
     * 是否为最后一条消息
     */
    @JsonProperty("isEnd")
    private Boolean isEnd;

    /**
     * 错误信息（仅 FAILED 状态时有值）
     */
    @JsonProperty("errorMsg")
    private String errorMsg;

    /**
     * 毫秒级时间戳
     */
    @JsonProperty("timestamp")
    private Long timestamp;
}

File: agent-code-platform/app/model/event/task_result_event.py
from enum import Enum

from pydantic import BaseModel, ConfigDict, Field


class TaskStatus(str, Enum):
    """任务状态"""

    PENDING = "PENDING"
    PROCESSING = "PROCESSING"
    SUCCESS = "SUCCESS"
    FAILED = "FAILED"
    WAITING_RETRY = "WAITING_RETRY"
    CANCELLED = "CANCELLED"
    STREAMING = "STREAMING"
    INTERRUPTED = "INTERRUPTED"


class MessageType(str, Enum):
    """消息类型"""

    AI = "ai"
    USER = "user"
    SYSTEM = "system"


class TaskResultEvent(BaseModel):
    """Kafka结果消息模型，发送到 task-result-topic"""

    task_id: str = Field(alias="taskId")
    app_id: int = Field(alias="appId")
    user_id: int = Field(alias="userId")
    status: TaskStatus = Field(alias="status")
    message_type: MessageType = Field(default=MessageType.AI, alias="messageType")
    content: str = Field(default="", alias="content")
    seq: int = Field(default=0, alias="seq")
    is_end: bool = Field(default=False, alias="isEnd")
    error_msg: str | None = Field(default=None, alias="errorMsg")
    timestamp: int = Field(alias="timestamp")

    model_config = ConfigDict(populate_by_name=True)
